From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 11 Oct 2018 18:12:54 +0000
Subject: appFolder: Don't block all shortcuts

App folder popups take a grab when opened, and as we don't pass any
particular pushModal() parameters, all keybindings are blocked. While
this makes sense for most keybindings that would interfere with the
popup interaction, others like volume/brightness keys or screenshots
can be allowed safely.

(cherry picked from commit 76117fd3063bcbbc8c8a0287c1ecf351d1a22039)

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/issues/648
Applied-upstream: 3.30.2, commit:54f8232a9339692c5307ff5220f83f27d5542622
---
 js/ui/appDisplay.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 67b796a..bfc6ec4 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -1462,7 +1462,9 @@ var AppFolderPopup = new Lang.Class({
         global.focus_manager.add_group(this.actor);
 
         source.actor.connect('destroy', () => { this.actor.destroy(); });
-        this._grabHelper = new GrabHelper.GrabHelper(this.actor);
+        this._grabHelper = new GrabHelper.GrabHelper(this.actor, {
+            actionMode: Shell.ActionMode.POPUP
+        });
         this._grabHelper.addActor(Main.layoutManager.overviewGroup);
         this.actor.connect('key-press-event', this._onKeyPress.bind(this));
     },
